pkg/registry/core/service/storage/rest.go

```diff
@@ -162,8 +218,44 @@ func (rs *REST) Export(ctx context.Context, name string, opts metav1.ExportOptio
 	return rs.services.Export(ctx, name, opts)
 }
 
+func (rs *REST) isExternalIPAM(ctx context.Context, service *api.Service) (bool, error) {
+	netName := service.Labels[arktosextensionsv1.NetworkLabel]
+	if netName == "" {
+		// choose the default network name if not specified
+		netName = "default"
+	}
+
+	network, err := rs.getNetwork(context.Background(), service.Tenant, netName)
```

Comment:
Why discard the ctx in the input param and create a new one?

Reply:
Actually, the tenant info should be available from ctx. It should be the same as the Tenant in service.

Reply:
the service context has service related info, which would lead to misleading error message (which seems derived based on context request info). Having a fresh context mitigate this problem. However, it makes sense if a subcontext can have new request info (get network cr) on top of the ctx param; a todo comment added as reminder for such consideration in future revisits.